nservicebus/messaging/uniformsession.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Uniform Session between message session and pipeline
+summary: Uniform Session allows to bridge the gap between the message session and the pipeline context and enables dependency injection.
+reviewed: 2017-10-27
+component: UniformSession
+---
+
+NServiceBus v6 and higher introduced a few design changes that how code that is using previous versions is structured. The [changes](/nservicebus/upgrades/5to6/moving-away-from-ibus.md) are:
+
+- In previous versions, the `IBus` interface was automatically registered in the IOC container. In Version 6, the new context-aware interfaces, for example, `IEndpointInstance`, `IMessageSession` and `IMessageHandlerContext`, etc., are not automatically registered in dependency injection.
+- In Versions 5 and below, when a custom component was registered in the container, the custom component had access to the `IBus` instance via dependency injection.
+- Clear separation of concerns between message operations outside the message handling pipeline (`IEndpointInstance` and `IMessageSession`) vs. inside the message handling pipeline (`IMessageHandlerContext`). Operations inside the message handling pipeline enlist in the transaction available and participate in the batching operations of the transport. Operations outside the pipeline don't have those capabilities. The duality of the `IBus` interface was removed by introducing separate explicit interfaces for each usage scenario.
```

Comment:
I think this part  (the last bullet point) is the biggest change and should be the first bullet point

Reply:
after the bullet points, we could add the finishing sentence like "For all details, visit the [NServiceBus Version 5 to 6 upgrade guide](url)."